### PR TITLE
[Kubernetes] Add support to 1.26 k8s version

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?
Adds support for Kubernetes 1.26.x version and update kind version to 0.17.0

## Related issues

- Relates to https://github.com/elastic/observability-dev/issues/2345

### CODE REVIEW

**Review by gsantoro — APPROVED** [review]

**Review by gizas — APPROVED** [review]

**Comment by tetianakravchenko** [comment]
@elastic/elastic-agent-control-plane could you please have a look?

Windows tests failure is not related to the changes in this PR, and for some reason `Squash and merge` button is inactive, either because to failing tests, or because of missing review from codeowners, could you please review/merge it?

**Review by cmacknz — APPROVED** [review]

**Comment by cmacknz** [comment]
Looks like we have some new flaky tests, I'll merge this and we will address the tests separately.

**Pull request merged**
